qgym/benchmarks/metrics/initial_mapping_metrics.py

```diff
@@ -7,12 +7,32 @@
 the lower the better.
 """
 
+from __future__ import annotations
+
+from collections import deque
+from collections.abc import Iterable
+from copy import deepcopy
+from typing import Protocol, cast, runtime_checkable
+
 import networkx as nx
 import numpy as np
-from numpy.typing import NDArray
+from numpy.typing import ArrayLike, NDArray
+
+from qgym.generators.graph import BasicGraphGenerator, GraphGenerator
+from qgym.utils.input_validation import check_string
+
 
+@runtime_checkable
+class InitialMappingMetric(Protocol):
+    def compute(self, interaction_graph: nx.Graph, mapping: ArrayLike) -> float: ...
 
-class InitialMappingSolutionQuality:
+
+@runtime_checkable
+class Mapper(Protocol):
+    def compute_mapping(self, interaction_graph: nx.Graph) -> NDArray[np.int_]: ...
```

Comment:
I understand that you need a Mapper for the whole benchmarking workflow, but I find the place in the code a bit unexpected.